### PR TITLE
	Updated sente component to new function names in 1.8

### PR DESCRIPTION
**Pull request by theasp**
This renames `new-channel-sockets` to `new-channel-socket-server` (the
old name still works too) and adds `new-channel-socket-client` for use
in ClojureScript.  Additionally, both will get `:handler` from
`:sente-handler` if an event handler is not given.

### CODE REVIEW

**Comment by danielsz** [comment]
This looks great. 

This is the first component that has a cljs counterpart. Some documentation/usage info is probably warranted. Maybe a test for the cljs part can hit that spot. Thoughts?

File: src/system/components/sente.cljc [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+   (map->ChannelSocketServer {:web-server-adapter web-server-adapter
+                              :handler event-msg-handler
+                              :options options})))
+
+;; Old fn name
+(def new-channel-sockets new-channel-socket-server)
+
+;; Sente does not support CLJ as a client yet
+#?(:cljs
+   (defrecord ChannelSocketClient [ring-ajax-post ring-ajax-get-or-ws-handshake ch-chsk chsk-send! connected-uids router path handler options]
```
Comment (danielsz) [inline src/system/components/sente.cljc]: Aren't many fields in` ChannelSocketClient` extraneous? Or am I missing something?

**Comment by theasp** [comment]
You are absolutely correct about the fields for `ChannelSocketClient`, they are different for the client and server.  I should be able to do a test in ClojureScript.  Do you have a preferred plugin for tests?  I've been using doo personally: https://github.com/bensu/doo

**Comment by theasp** [comment]
Added a test case for sente-client using doo.  Let me know if you'd like anything changed.

**Comment by danielsz** [comment]
Awesome. I'll take a look and will let you know if there's something else. Thank you!

**Comment by danielsz** [comment]
Merged and deployed. Thanks, @theasp!

I'm curious about something, how extensively do you use components in your cljs projects? Any chance to have a look at a typical `system.cljs` of yours? I'd like to gain insight about how `system` is used on the client side. Thanks!

**Comment by theasp** [comment]
This is my first project using components in cljs, but so far I am liking it.  

My current (client) project looks like this:
```clojure
(defonce current-system (atom nil))
(defonce state (atom nil))

(def config
  {:web-root    "/"
   :development true})

(defn system []
  (component/system-map
   :state state
   :config config

   ;; :sente is the only bit I use from system on the client 
   :sente-handler {:handler event-msg-handler}
   :sente (-> (sente/new-channel-socket-client "/chsk")
              (component/using [:sente-handler]))

   ;; bidi routes inside the app, so that "/app/foo" shows the foo view.
   :app-routes (routes/new-app-routes)

   ;; bidi routes outside the app, on the webserver.  Used to generate URLs to resources
   :web-routes (routes/new-web-routes)

   ;; Functions to manipulate the state for the client.  Will morph into local storage later.
   :memory-store (memory-store/new-store)

   ;; The same functions as memory-store, but accessing the server through sente.
   ;; Does not exist yet
   ;;:sente-store (sente-store/new-store)

   ;; Magic to keep memory-store and sente-store in sync
   ;;:synchronizer (synchronizer/new-synchronizer)

   ;; This is actually the top level component for reagent, uses state to choose the view
   :views (views/new-views)

   ;; Generic component for reagent, uses views.  Could easily be part of system.
   :renderer (renderer/new-renderer)

   ;; Generic component for pushy.  Updates state when the URL of the browser changes, using app-routes.  Also easily added to system.
   :url-history (url-history/new-url-history)))

;; figwheel calls this when something is updated
(defn re-render []
  (when-let [renderer (:renderer @current-system)]
    (debugf "Re-render triggered")
    (renderer/render renderer)))

(defn start! []
  (reset! current-system (component/start (prod-system))))

(defn stop! []
  (when @current-system
    (component/stop @current-system))
  (reset! current-system nil))

(defn restart! []
  (stop!)
  (start!))

(defn init! []
  (start!))
```

I am considering switching the server from Clojure to ClojureScript (memory issues...), and I imagine it's components will look very similar to a typical Clojure webserver

**Pull request merged**